src/ir/item.rs

```diff
@@ -346,6 +366,10 @@ pub struct Item {
     parent_id: ItemId,
     /// The item kind.
     kind: ItemKind,
+    /// Detect cycles when determining if we can derive debug/copy or not, and
+    /// avoid infinite recursion.
+    detect_derive_debug_cycle: Cell<bool>,
```

Comment:
I think the real fix is https://github.com/servo/rust-bindgen/issues/536. That will not only remove the need for these cells, but also make it so we can generate this stuff for everything all at once rather than re-walking the same sub graphs repeatedly when asking these questions for similar-but-distinct types.

I'll file individual issues for each analysis and offer to mentor them. They would be good bugs for people who have contributed before and might be interested in more.